### DISCUSSION TITLE
Close Database Connection On User Leave

### DISCUSSION THREAD

**Discussion by ByteAtATime** [sveltejs/kit#8489] [Q&A]
I currently use the `pg` node package to connect to the database. Is there a way that I can close this connection on the server side when the user leaves? I’m looking for something like `load()` but for when the page unloads.

**Comment by teemingc** [comment]
Perhaps a combination of `beforeNavigate()` and checking `navigation.type` in a root layout to close the connection.

`+layout.svelte`
```svelte
<script>
  import { beforeNavigate } from '$app/navigation';
  
  beforeNavigate(({ type }) => {
    if (type === 'leave') {
      // close db connection
    }
  })
</script>
```

https://kit.svelte.dev/docs/modules#$app-navigation-beforenavigate
https://kit.svelte.dev/docs/types#public-types-navigation

**Reply by pevey** [reply]
You might consider instead using one db connection and keeping it open. Establish the connection in a small utility lib, and then import that wherever needed throughout your app.
